zulip/zulip/__init__.py

```diff
@@ -356,6 +356,7 @@ def __init__(
         insecure: Optional[bool] = None,
         client_cert: Optional[str] = None,
         client_cert_key: Optional[str] = None,
+        legacy_exception_handling: Optional[bool] = False,
```

Comment:
I don’t see any need for an option to request the old behavior. Any caller that would be modified to use it could also be modified to handle exceptions itself.